ethereum/trie/src/main/java/org/hyperledger/besu/ethereum/trie/SimpleMerklePatriciaTrie.java

```diff
@@ -99,8 +101,22 @@ public void commit(final NodeUpdater nodeUpdater) {
   }
 
   @Override
-  public void visitAll(final Consumer<Node<V>> visitor) {
-    root.accept(new AllNodesVisitor<>(visitor));
+  public void visitAll(final Consumer<Node<V>> nodeConsumer) {
+    root.accept(new AllNodesVisitor<>(nodeConsumer));
+  }
+
+  @Override
+  public CompletableFuture<Void> visitAll(
+      final Consumer<Node<V>> nodeConsumer, final ExecutorService executorService) {
+    nodeConsumer.accept(root);
```

Comment:
Alternatively, I'd be tempted to just expose `getRoot()` on the `Trie` interface, and move this logic into the `Pruner`...

Reply:
I kinda like it being on the Trie because we're adding other traversal methods there. It becomes one place to look various types of trie iteration. Not sure if I understood what you were suggesting so does that make sense?